src/System.Private.ServiceModel/src/System/IdentityModel/Claims/X509CertificateClaimSet.cs

```diff
@@ -267,6 +268,32 @@ public override IEnumerable<Claim> FindClaims(string claimType, string right)
             }
         }
 
+        private static string[] GetDnsFromExtensions(X509Certificate2 cert)
+        {
+            foreach (X509Extension ext in cert.Extensions)
+            {
+                // Extension is SAN or SAN2
+                if (ext.Oid.Value == "2.5.29.7" || ext.Oid.Value == "2.5.29.17")
+                {
+                    string asnString = ext.Format(true);
+                    if (string.IsNullOrEmpty(asnString))
+                    {
+                        return new string[0];
```

Comment:
I may take this as an additional comment after this is merged; this current commit brings us into line with what Desktop looks like